backends/ebpf/psa/externs/ebpfPsaRegister.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright 2022-present Orange
+Copyright 2022-present Open Networking Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "ebpfPsaRegister.h"
+#include "backends/ebpf/psa/ebpfPsaControl.h"
+
+namespace EBPF {
+
+EBPFRegisterPSA::EBPFRegisterPSA(const EBPFProgram *program,
+                                 cstring instanceName, const IR::Declaration_Instance *di,
+                                 CodeGenInspector *codeGen) : EBPFTableBase(program,
+                                                                            instanceName,
+                                                                            codeGen) {
+    CHECK_NULL(di);
+    if (!di->type->is<IR::Type_Specialized>()) {
+        ::error(ErrorType::ERR_MODEL, "Missing specialization: %1%", di);
+        return;
+    }
+    auto ts = di->type->to<IR::Type_Specialized>();
```

Comment:
Didn't you create some helper functions to extract arguments from specialized types?

Reply:
`EBPFObject::getSpecializedTypeName` only gets a type name, we need to extract arguments here.